### PR TITLE
[ANCHOR-357] SEP-6: Implement deposit-exchange

### PR DESCRIPTION
**Pull request by philipliu**
### Description

This implements the GET `deposit-exchange` endpoint. There's a lot of duplicated code that could be shared with `deposit` and `withdraw`/`withdraw-exchange`. Once the integration and end-to-end tests have been implemented, I will focus on cleaning up the code.

### Context

SEP-6 implementation.

### Testing

`./gradlew test`

### Known limitations

Integration and end-to-end tests are missing, they will be added in another PR.

### CODE REVIEW

**Comment by philipliu** [comment]
Github does not support stacked PRs unfortunately. This is based on withdraw-exchange PR so it might be a bit difficult to review without that being checked in first.

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java [hunk 1]
```diff
@@ -119,6 +119,91 @@ public StartDepositResponse deposit(Sep10Jwt token, StartDepositRequest request)
...
+    txnStore.save(txn);
+
+    eventSession.publish(
+        AnchorEvent.builder()
+            .id(UUID.randomUUID().toString())
+            .sep("6")
+            .type(AnchorEvent.Type.TRANSACTION_CREATED)
+            .transaction(TransactionHelper.toGetTransactionResponse(txn, assetService))
+            .build());
+
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:200]: 💡 Increment metrics counter.

**Review by lijamie98 — COMMENTED** [review]
LGTM. Please feel free to merge after the comment is taken care of.

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java (hunk 1 above)
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:200]: Adding it as part of ANCHOR-483.

**Review by lijamie98 — APPROVED** [review]

**Pull request merged**